### ISSUE TITLE
Bug: Autofill Issue with native filters 

### ISSUE THREAD

**Issue by vigodeltoro** [apache/superset#23903]
Hi over there :)

I guess I found a bug concerning the filter boxes. Maybe anybody will be so kind to have a look, I would really appreciate that.

Thanks in advance..

All filter values are chosen if a filter value is directly inserted via keyboard and not chosen by list of values.

#### How to reproduce the bug

Open up a dashboard with native filters and some filter values inside the filter. Don't select a value from the list, type the complete value of one of the filter values into the filter value field and press "enter". I expect that only that special value selected. But instead of this all values are selected.

### Expected results

I expect that only that special value selected. But instead of this all values are selected.

### Actual results

All values are selected.

#### Screenshots

![image](https://user-images.githubusercontent.com/8578198/235654558-1fc19f33-4d17-42e8-a403-164ab62db400.png)

![image](https://user-images.githubusercontent.com/8578198/235654591-35c5bca3-a440-4d1e-8c90-757c80bf7eec.png)

### Environment

(please complete the following information):

- browser type and version: (Safari Version 16.3 (18614.4.6.1.6), Firefox 111.01
- superset version: 2.1.0
- python version: `3.8.16`
- node.js version: ? ( we are using the superset docker container (at dockerhub apache/superset/2.1.0)
- any feature flags active: ( EMBEDDED_SUPERSET,  DASHBOARD_RBAC, DASHBOARD_CROSS_FILTERS, VERSIONED_EXPORT)
### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [x ] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [ x] I have reproduced the issue with at least the latest released version of superset.
- [ x] I have checked the issue tracker for the same issue and I haven't found one similar.

Thanks a lot for your help.. 
best regards

**Comment by LucasGabrielB** [comment]
Having the same problem here.

**Comment by itziarmj** [comment]
Yes, it selects the first option in the list, that is the new "select all" option. 
I think the behaviour is that it selects the slightly highlighted option in the list, that by default is the first one. 
This is quite confusing for the users, that are expecting just the inserted text to be selected.
In case there is no matching value in the list and it is empty, together with the inserted text, it also enters a second option with empty value.

![image](https://github.com/apache/superset/assets/101407671/d63ac034-ce96-46fb-b6d5-0bed1dcde5d0)

**Comment by vigodeltoro** [comment]
Hi there.. 
I checked it in my test env.. no, 3.x fix the issue.. thanks a lot :)
👍

**Comment by rusackas** [comment]
Closing as resolved. Thanks for the verification @vigodeltoro :)
